aspnetcore/blazor/fundamentals/dependency-injection.md

```diff
@@ -317,15 +317,19 @@ public IMyService MyService { get; set; }
 
 ## Utility base component classes to manage a DI scope
 
-In ASP.NET Core apps, scoped services are typically scoped to the current request. After the request completes, any scoped or transient services are disposed by the DI system. Server-side, the request scope lasts for the duration of the client connection, which can result in transient and scoped services living much longer than expected. Client-side, services registered with a scoped lifetime are treated as singletons, so they live longer than scoped services in typical ASP.NET Core apps.
+In non-Blazor ASP.NET Core apps, scoped and transient services are typically scoped to the current request. After the request completes, scoped and transient services are disposed by the DI system.
+
+In interactive server-side Blazor apps, the request scope lasts for the duration of the circuit (the SignalR connection between the client and server), which can result in scoped and transient services living much longer than expected. Therefore, direct use of scoped services should be avoided, and transient services shouldn't be registered or used at all. An alternative approach based on the <xref:Microsoft.AspNetCore.Components.OwningComponentBase> type is described later in this section.
```

Comment:
> direct use of scoped services should be avoided

Was this guidance that we had before? Scoped services can be useful when using limiting the scope of a stateful service to a circuit/user (e.g., if you need to manage user-specific data).

Reply:
I see what you mean. It needs more work ... I think it really should say something like ...

> ... can result in scoped and transient services living much longer than expected if you only want the service to exist for the lifetime of a single component. Therefore, direct use of scoped services for single component lifetimes should be avoided, and transient services shouldn't be registered or used at all.

... does that seem better? ... more specific to what this is leading to for the use of `OwningComponentBase`.

Reply:
I'll drop that as a full suggestion here ...

```suggestion
In interactive server-side Blazor apps, the request scope lasts for the duration of the circuit (the SignalR connection between the client and server), which can result in scoped and transient services living much longer than expected for the lifetime of a single component. Therefore, direct use of scoped services for single component lifetimes should be avoided, and transient services shouldn't be registered or used at all. An alternative approach based on the <xref:Microsoft.AspNetCore.Components.OwningComponentBase> type is described later in this section.
```

Reply:
Yes, that looks great!